.github/workflows/merge.yaml

```diff
@@ -0,0 +1,98 @@
+      with:
+        fetch-depth: 0
+
+    - name: Checkout release branch
+      uses: actions/checkout@v2
+      with:
+        ref: 'v1.0.1'
+        path: './release'
+
+    - name: Check for relevant changes
+      if: github.event_name == 'push'
+      uses: dorny/paths-filter@v2
+      id: release
+      with:
+        list-files: shell
+        filters: |
+          updated:
+            - added|modified: [ '*.csl', '*.xml' ]
+          deleted:
+            - deleted: [ '*.csl', '*.xml' ]
+
+    - name: Changed files
+      if: github.event_name == 'push'
+      run: |
+        echo updated: ${{ steps.release.outputs.updated_files }}
+        echo deleted: ${{ steps.release.outputs.deleted_files }}
+
+    - name: Set up Ruby
+      if: github.event_name == 'workflow_dispatch' || (github.event_name == 'push' && (steps.release.outputs.updated == 'true' || steps.release.outputs.deleted == 'true'))
+      uses: ruby/setup-ruby@v1
+      with:
+        ruby-version: 2.7.1
+    - name: but use cache to speed that up
+      if: github.event_name == 'workflow_dispatch' || (github.event_name == 'push' && (steps.release.outputs.updated == 'true' || steps.release.outputs.deleted == 'true'))
+      uses: actions/cache@v2
+      with:
+        path: vendor/bundle
+        key: ${{ runner.os }}-gems-${{ hashFiles('**/Gemfile.lock') }}
+        restore-keys: |
+          ${{ runner.os }}-gems-
+    - name: Bundle install
+      if: github.event_name == 'workflow_dispatch' || (github.event_name == 'push' && (steps.release.outputs.updated == 'true' || steps.release.outputs.deleted == 'true'))
+      run: |
+        bundle config path vendor/bundle
+        bundle update sheldon --jobs 4 --retry 3
+
+    - name: Populate new branch
+      run: bundle exec sheldon --token=$GITHUB_TOKEN --verbose --populate release
+      if: github.event_name == 'workflow_dispatch'
+
+    - name: update the timestamps and add the changes
+      run: bundle exec sheldon --token=$GITHUB_TOKEN --verbose --release release ${{ steps.release.outputs.updated_files }}
+      if: github.event_name == 'push' && steps.release.outputs.updated == 'true'
+
+    - name: delete deleted files
+      run: cd release && git rm ${{ steps.release.outputs.deleted_files }}
+      if: github.event_name == 'push' && steps.release.outputs.deleted == 'true'
+
+    - uses: stefanzweifel/git-auto-commit-action@v4
+      with:
+        repository: 'release'
+        commit_message: Releasing ${{ steps.release.outputs.updated_files }} ${{ steps.release.outputs.deleted_files }}
+      if: github.event_name == 'push' && (steps.release.outputs.updated == 'true' || steps.release.outputs.deleted == 'true')
+
+    - uses: stefanzweifel/git-auto-commit-action@v4
+      with:
+        repository: 'release'
+        commit_message: ${{ github.event.inputs.commit_message }}
+      if: github.event_name == 'workflow_dispatch'
+
+    - name: ping distribution-updater
```

Comment:
Could you clarify that this is pinging Zotero more generally (styles page, Zotero client updates, etc.), not just the distribution-updater? I'd like to avoid a situation where the distribution-updater is removed in the future and someone forgets that this is also updating styles across the entire Zotero ecosystem.

Reply:
> I'd like to avoid a situation where the distribution-updater is removed in the future and someone forgets that this is also updating styles across the entire Zotero ecosystem.

@dstillman, I think you told me before, but can you confirm that Zotero doesn't rely on https://github.com/citation-style-language/styles-distribution/ and https://github.com/citation-style-language/distribution-updater at all? I'm not sure how popular https://github.com/citation-style-language/styles-distribution/ is, but I think we'd want to deprecate it rather quickly and post a notification that people should now tune in at https://github.com/citation-style-language/styles/tree/v1.0.1 to get their vetted CSL 1.0.1 styles.

(cc @bwiernik, @adam3smith)

Reply:
We don't use styles-distribution, no. We just need the 8826 ping.

Reply:
When will >1.0.1 styles start being committed to `master`, though? If that's going to happen before Zotero supports them, we would need to make some changes on our end to make sure we weren't pushing out incompatible styles.

On that subject, @retorquere, it would be better to include the actual branch name for `branch`, not just hard-code "master".

Reply:
> When will 1.1 styles start being committed to `master`, though?

@bwiernik might have an idea (I've not been very involved in the 1.0.2 and 1.1 release development). I did just propose a branch scheme for maintaining a 1.0.x dev branch. See https://github.com/citation-style-language/styles/pull/5126#issuecomment-734629766.

Reply:
> On that subject, @retorquere, it would be better to include the actual branch name for branch, not just hard-code "master".

No problem with that, but I thought `master` was the outcome of the earlier discussion. What is "the actual branch name" in this context? The `v1.0.1` release branch?

Reply:
We will probably finish up the 1.1 schema and spec by January. From there, we'd probably want processors to update before we start to push 1.1 styles. Perhaps then would be a good time to notify clients and users to update their paths?

Reply:
> What is "the actual branch name" in this context?

Whatever commit the branch is on, the same as with the Travis requests.

If GH actions don't provide a variable for the branch, I think it would be the result of `git rev-parse --abbrev-ref HEAD`.

Reply:
> Whatever branch the commit is on, the same as with the Travis requests.

which means "whatever branch *in the `release` directory is on", which in turn would translate to `v1.0.1` currently, correct? Because from the pov of the action, "the branch the commit is on" is always "master".

> If GH actions don't provide a variable for the branch, I think it would be the result of `git rev-parse --abbrev-ref HEAD`.

We have a var for the branch that the action is running on, and the `v1.0.1` branch we check out ourselves, so I can stuff that into a var for reuse.

Reply:
Looks like @dstillman 's original concern here hasn't actually been addressed, right? so `ping zotero-updater` as a name for this?
And then using a variable for the branch name also isn't done yet, correct?

Reply:
If the locales and styles repos always have the same current-release version, https://github.com/citation-style-language/styles/pull/5141/files would take care of cross-applying changes to the workflows.

Reply:
@retorquere I'm not sure how that answers either of Sebastian's questions?

Reply:
Since [this PR](https://github.com/citation-style-language/styles/pull/5141) would cause any change to the `styles` workflows to be copied over to the `locales` workflows, the line would automatically become "[ping Zotero servers](https://github.com/citation-style-language/styles/blob/56fde4bf291adb0e68482271310c644a5d5dd913/.github/workflows/merge.yaml#L102)".

Reply:
got it, thanks